pkg/apiserver/statement/models.go

```diff
@@ -96,57 +103,103 @@ type Model struct {
-	AggPlanCount             int    `json:"plan_count" agg:"COUNT(DISTINCT plan_digest)"`
+	AggPlanCount             int    `json:"plan_count" agg:"COUNT(DISTINCT plan_digest)" related:"plan_digest"`
 	AggPlan                  string `json:"plan" agg:"ANY_VALUE(plan)"`
 	AggPlanDigest            string `json:"plan_digest" agg:"ANY_VALUE(plan_digest)"`
+	// RocksDB
+	AggMaxRocksdbDeleteSkippedCount uint `json:"max_rocksdb_delete_skipped_count" agg:"MAX(max_rocksdb_delete_skipped_count)"`
+	AggAvgRocksdbDeleteSkippedCount uint `json:"avg_rocksdb_delete_skipped_count" agg:"CAST(SUM(exec_count * avg_rocksdb_delete_skipped_count) / SUM(exec_count) as SIGNED)"`
+	AggMaxRocksdbKeySkippedCount    uint `json:"max_rocksdb_key_skipped_count" agg:"MAX(max_rocksdb_key_skipped_count)"`
+	AggAvgRocksdbKeySkippedCount    uint `json:"avg_rocksdb_key_skipped_count" agg:"CAST(SUM(exec_count * avg_rocksdb_key_skipped_count) / SUM(exec_count) as SIGNED)"`
+	AggMaxRocksdbBlockCacheHitCount uint `json:"max_rocksdb_block_cache_hit_count" agg:"MAX(max_rocksdb_block_cache_hit_count)"`
+	AggAvgRocksdbBlockCacheHitCount uint `json:"avg_rocksdb_block_cache_hit_count" agg:"CAST(SUM(exec_count * avg_rocksdb_block_cache_hit_count) / SUM(exec_count) as SIGNED)"`
+	AggMaxRocksdbBlockReadCount     uint `json:"max_rocksdb_block_read_count" agg:"MAX(max_rocksdb_block_read_count)"`
+	AggAvgRocksdbBlockReadCount     uint `json:"avg_rocksdb_block_read_count" agg:"CAST(SUM(exec_count * avg_rocksdb_block_read_count) / SUM(exec_count) as SIGNED)"`
+	AggMaxRocksdbBlockReadByte      uint `json:"max_rocksdb_block_read_byte" agg:"MAX(max_rocksdb_block_read_byte)"`
+	AggAvgRocksdbBlockReadByte      uint `json:"avg_rocksdb_block_read_byte" agg:"CAST(SUM(exec_count * avg_rocksdb_block_read_byte) / SUM(exec_count) as SIGNED)"`
 	// Computed fields
 	RelatedSchemas string `json:"related_schemas"`
 }
 
 var cachedAggrMap map[string]string // jsonFieldName => aggr
 
-func getAggrMap() map[string]string {
+func getAggrMap() (map[string]string, error) {
 	if cachedAggrMap == nil {
 		t := reflect.TypeOf(Model{})
 		fieldsNum := t.NumField()
 		ret := map[string]string{}
+
 		for i := 0; i < fieldsNum; i++ {
 			field := t.Field(i)
 			jsonField := strings.ToLower(field.Tag.Get("json"))
-			if agg, ok := field.Tag.Lookup("agg"); ok {
+			rf, ok := field.Tag.Lookup(RelatedTag)
+			var rfs []string
+			if ok {
+				rfs = strings.Split(rf, ",")
+			} else {
+				rfs = append(rfs, jsonField)
+			}
+
+			// Filtering fields that are not in the table fields
+			verified, err := verifiedAggr(rfs)
+			if err != nil {
+				return nil, err
+			}
+
+			if agg, ok := field.Tag.Lookup("agg"); ok && verified {
 				ret[jsonField] = fmt.Sprintf("%s AS %s", agg, gorm.ToColumnName(field.Name))
 			}
 		}
 		cachedAggrMap = ret
 	}
-	return cachedAggrMap
+	return cachedAggrMap, nil
+}
+
+func verifiedAggr(relatedFields []string) (bool, error) {
```

Comment:
Please give some short comments about this function, since it's name is not very easy to be understood..